data/sovereign/operationData.go

```diff
@@ -0,0 +1,41 @@
+package sovereign
+
+import "math/big"
+
+// Operation holds the data needed for a bridge transfer
+type Operation struct {
+	Address      []byte
+	Tokens       []EsdtToken
+	TransferData *TransferData
+}
+
+// EsdtToken holds the token data
+type EsdtToken struct {
+	Identifier []byte
+	Nonce      uint64
+	Data       EsdtTokenData
+}
+
+// EsdtTokenType holds the type of the esdt token
+type EsdtTokenType uint32
+
+const (
```

Comment:
Do you really need these for anything?
We already have consts for fungible and non fungible anyway in core

Reply:
```go
// ESDTType defines the possible types in case of ESDT tokens
type ESDTType uint32

const (
	// Fungible defines the token type for ESDT fungible tokens
	Fungible ESDTType = iota
	// NonFungible defines the token type for ESDT non fungible tokens
	NonFungible
)
```

Reply:
Those 2 are the only ones you need when deciding how to encode them for incoming operations